bridges/GQMagazineBridge.php

```diff
@@ -101,7 +101,7 @@ private function loadFullArticle($uri){
 			// List the CSS classes of that div
 			$classes = $div->class;
 			// I can't directly lookup that class since GQ since to generate random names like "ArticleBodySection-fkggUW"
-			if(strpos($classes, 'ArticleBodySection') !== false) {
+			if(strpos($classes, 'ArticleWrapper') !== false) {
```

Comment:
The parser supports CSS selectors like `$html->find('[class*="ArticleWrapper"]')` or `$html->find('[class^="ArticleWrapper"]')` to find elements that contain or start with "ArticleWrapper".

Maybe that is useful here?

Reply:
Well, in fact, if you take a look at the HTML code of any GQ article page, you'll find that classes are things like that

	<div class="ArticleWrapper-liwRXn koIApb"><div class="ArticleBorderTop-hqkxUY fdLxUX"></div><div class="Layout-cEduSl FlexLayout-bTxyMW gnHBxh"><section class="MainContentWrapper-kLTFQs ccvTRF"><div class="AdWrapperRow-ewuCWT knVSff"><div data-test-id="AdBannerWrapper" class="Hide-fmACZo AdBannerWrapper-fbkyjt jUzCdk"><div class="sc-bwzfXH bGgvvP"><div style="height:auto" class="sc-bdVaJa BcOix"><div class="ad " data-ad-unit-path="/5574/fr-gq-cp/sexe/sexactu/article/display-pos-1" aria-hidden="true" data-size-mapping="[{&quot;viewport&quot;:[0,0],&quot;slot&quot;:[[2,2],[300,50],[300,100],[300,250],[320,50],[320,100],[320,160],[336,280],[768,768]],&quot;initialHeight&quot;:250},{&quot;viewport&quot;:[728,0],&quot;slot&quot;:[[2,2],[728,90],[1920,1080]],&quot;initialHeight&quot;:90},{&quot;viewport&quot;:[900,0],&quot;slot&quot;:[],&quot;initialHeight&quot;:0}]" data-kv-pairs="[[&quot;display-pos&quot;,1]]" id="ad-21"></div></div></div></div></div><div class="PullQuoteWrapper-dlqknE gQqVvB"><blockquote class="BaseText-dZTLdJ BaseTitle-inZYgN Headline-Sqhiv Headline5-fYuCZb CommonBlockquote-bynhSh Blockquote-hZqgCM fmOqLZ"><p class="Body-MnRsT Paragraph-cbzGIL bAebkj">"Pour Christine Castelain Meunier, l’arme se retourne contre celle qui la tend “car cela consiste à rentrer dans le jeu de la compétition masculine”. C’est également l’avis de Victoire Tuaillon, qui depuis plus d’un an passe les masculinités au crible dans son podcast Les Couilles sur la table. À l’écouter, impossible de se dire féministe si l’on qualifie un mec de petite bite: “C’est tactiquement stupide et rétrograde: cela revient à traiter une femme de vieille peau”. Loin de couper la chique aux mâles alpha qui sacralisent les vingt trois centimètres de Rocco Siffredi, ces mots-là perpétuent un imaginaire hétéronormé nourri par les codes du X: celui de la “performance” et de la sacro-sainte pénétration, “l’idée que la valeur d’un homme et de son rapport sexuel dépendent de la grosseur de ses organes génitaux”.</p></blockquote></div><p class="Body-MnRsT Paragraph-cbzGIL bAebkj">- Cette semaine, on a redécouver<a href="https://www.terrafemina.com/article/vontouring-mais-a-quoi-sert-cette-nouvelle-tendance-de-contouring-du-vagin_a280139/1" rel="nofollow">t le vontouring</a> (le contouring du vagin), dont on parlait déjà en 2015 :</p><div class="TeadRow-jWLBVL fbxQZE"><div class="sc-bwzfXH bGgvvP"><div><div style="height:auto" class="sc-bdVaJa eDCunu"><div class="ad " data-ad-unit-path="/5574/fr-gq-cp/sexe/sexactu/article/outstream" aria-hidden="true" data-is-out-of-page="true" data-kv-pairs="[[&quot;out-of-page&quot;,&quot;outstream&quot;]]" id="ad-22"></div></div><div class="teads"></div></div></div></div><div class="PullQuoteWrapper-dlqknE gQqVvB"><blockquote class="BaseText-dZTLdJ BaseTitle-inZYgN Headline-Sqhiv Headline5-fYuCZb CommonBlockquote-bynhSh Blockquote-hZqgCM fmOqLZ"><p class="Body-MnRsT Paragraph-cbzGIL bAebkj">Ce procédé, qui s'appelle en réalité Intima Protégé, vise à corriger l'aspect de l'extérieur de votre vagin et de votre vulve en utilisant des ultrasons qui stimulent la production de collagène en appliquant une chaleur intense sur l'ouverture du vagin. C'est un traitement sans douleur qui ne nécessite ni intervention chirurgicale, ni injections, ni incisions. Bref, c'est censé être carrément moins invasif que les labioplasties et autres opérations chirurgicales de la vulve. Ce n'est pas comme le Vajacula par exemple, qui nécessite qu'on vous injecte du sang dans les muqueuses.</p></blockquote></div><p class="Body-MnRsT Paragraph-cbzGIL bAebkj">- Chez Slate, <a href="http://www.slate.fr/story/172806/hommes-desir-enfant-injonctions-paternite-maternite" rel="nofollow">on parle des hommes qui ne veulent pas d'enfants</a> :</p><p class="Body-MnRsT Paragraph-cbzGIL bAebkj">Faut-il traiter un mec de petite bite ? <a href="http://cheekmagazine.fr/societe/insulte-mec-petite-bite/" rel="nofollow">Le meilleur article de la semaine est à lire chez Cheeks</a>, et il est super bien documenté :</p><div class="AdWrapperRow-ewuCWT knVSff"><div data-test-id="AdBannerWrapper" class="Hide-fmACZo AdBannerWrapper-fbkyjt jUzCdk"><div class="sc-bwzfXH bGgvvP"><div style="height:auto" class="sc-bdVaJa eWSIPO"><div class="ad " data-ad-unit-path="/5574/fr-gq-cp/sexe/sexactu/article/display-pos-2" aria-hidden="true" data-size-mapping="[{&quot;viewport&quot;:[0,0],&quot;slot&quot;:[[2,2],[300,50],[300,100],[300,250],[300,600],[320,50],[320,100],[320,160],[336,280]],&quot;initialHeight&quot;:600},{&quot;viewport&quot;:[900,0],&quot;slot&quot;:[],&quot;initialHeight&quot;:0}]" data-kv-pairs="[[&quot;display-pos&quot;,2]]" id="ad-23"></div></div></div></div></div><div class="PullQuoteWrapper-dlqknE gQqVvB"><blockquote class="BaseText-dZTLdJ BaseTitle-inZYgN Headline-Sqhiv Headline5-fYuCZb CommonBlockquote-bynhSh Blockquote-hZqgCM fmOqLZ"><p class="Body-MnRsT Paragraph-cbzGIL bAebkj">En France, 5% de la population ne souhaitent pas d’enfants selon l’Institut national d’études démographiques (Ined). Les raisons sont-elles identiques pour les femmes et les hommes? Pour Charlotte Debest, ce sont les mêmes: la liberté d’entreprendre, l’indépendance, l’envie de rester mobile. «Un enfant entraîne de la dépendance et crée un lien inaliénable, c’est donc aussi des contraintes.»</p></blockquote></div><p class="Body-MnRsT Paragraph-cbzGIL bAebkj">- Amour infini pour Agnès Giard, <a href="http://sexes.blogs.liberation.fr/2019/01/29/pourquoi-dit-erection-pour-un-gratte-ciel/" rel="nofollow">qui parle d'érections en architecture</a> (on y apprend que ces bâtiments sont une plaie à entretenir, soit dit en passant) :</p><div class="PullQuoteWrapper-dlqknE gQqVvB"><blockquote class="BaseText-dZTLdJ BaseTitle-inZYgN Headline-Sqhiv Headline5-fYuCZb CommonBlockquote-bynhSh Blockquote-hZqgCM fmOqLZ"><p class="Body-MnRsT Paragraph-cbzGIL bAebkj">Le mot «érection» vient du latin erectio («action d’élever, de dresser»), dérivé du verbe erigere («mettre droit, ériger»). Son étymologie même renvoie à l’idée de l’architecture. En grec, l’idée du phallus comme construction est aussi prégnante. Le mot «érection» se traduit en grec par στύση (stisi) qui a un rapport avec στύλος (stylos) –“pilier”– dérivé du verbe ἵστημι (istamai) : “se tenir debout”, “supporter”. Stylos désigne donc une chose longue et étroite, cylindrique ou rectangulaire qui tient ou supporte quelque chose au bout : une colonne, par exemple. Ou un pistil.</p></blockquote></div><p class="Body-MnRsT Paragraph-cbzGIL bAebkj">- Télérama pose les bonnes questions : <a href="https://www.telerama.fr/medias/instagram-serait-il-le-nouvel-espace-de-la-liberation-sexuelle,n6104551.php" rel="nofollow">Instagram serait-il le nouvel espace de la libération de la parole sexuelle</a> ? (Je découvre au passage que je suis complètement larguée en libération de la parole sexuelle sur Instagram.)</p><div class="PullQuoteWrapper-dlqknE gQqVvB"><blockquote class="BaseText-dZTLdJ BaseTitle-inZYgN Headline-Sqhiv Headline5-fYuCZb CommonBlockquote-bynhSh Blockquote-hZqgCM fmOqLZ"><p class="Body-MnRsT Paragraph-cbzGIL bAebkj">« Alors, t’as joui ? » Ce symbole du mystère persistant de l’orgasme féminin est aussi le nom d’un projet Instagram qui, ces derniers mois, a fait parler de lui. Ce compte partage des témoignages de femmes tantôt désolants, tantôt révoltants sur leurs expériences sexuelles avec les hommes. Il n’est pas le seul à s’être lancé sur ce terrain. Depuis le printemps dernier, les comptes qui parlent de sexe fleurissent : Jouissance club, Clitrevolution, Tu bandes ?, Merci beaucul… Féministes, décomplexés et inclusifs, ils sont à mille lieues des « conseils » culpabilisants rabâchés par la presse féminine, et ça fait du bien…</p></blockquote></div><p class="Body-MnRsT Paragraph-cbzGIL bAebkj">- Il faut le rappeler, <a href="https://www.rtbf.be/info/societe/detail_comment-les-images-de-corps-sexualises-nous-font-tolerer-le-harcelement?id=10125607" rel="nofollow">l’exposition aux images sexualisées favorise le sexisme</a> (les féministes ne sont pas puritaines, elles sont pragmatiques) :</p><div class="AdWrapperRow-ewuCWT knVSff"><div data-test-id="AdBannerWrapper" class="Hide-fmACZo AdBannerWrapper-fbkyjt jUzCdk"><div class="sc-bwzfXH bGgvvP"><div style="height:auto" class="sc-bdVaJa eWSIPO"><div class="ad " data-ad-unit-path="/5574/fr-gq-cp/sexe/sexactu/article/display-pos-3" aria-hidden="true" data-size-mapping="[{&quot;viewport&quot;:[0,0],&quot;slot&quot;:[[2,2],[300,50],[300,100],[300,250],[300,600],[320,50],[320,100],[320,160],[336,280]],&quot;initialHeight&quot;:600},{&quot;viewport&quot;:[900,0],&quot;slot&quot;:[],&quot;initialHeight&quot;:0}]" data-kv-pairs="[[&quot;display-pos&quot;,3]]" id="ad-24"></div></div></div></div></div><div class="PullQuoteWrapper-dlqknE gQqVvB"><blockquote class="BaseText-dZTLdJ BaseTitle-inZYgN Headline-Sqhiv Headline5-fYuCZb CommonBlockquote-bynhSh Blockquote-hZqgCM fmOqLZ"><p class="Body-MnRsT Paragraph-cbzGIL bAebkj">"Par exemple, l’une de nos études réalisées à l’ULB indique que l’exposition à des vidéoclips dans lesquels des chanteuses sont représentées de façon sexualisée modifie les attitudes à l’égard du harcèlement sexuel. Les participants de notre étude qui furent exposés à ces vidéoclips avaient plus tendance à blâmer une femme victime de harcèlement sexuel que les participants ayant visionné des vidéoclips représentant ces mêmes chanteuses de façon non-sexualisée".</p></blockquote></div><p class="Body-MnRsT Paragraph-cbzGIL bAebkj"><strong>En anglais :</strong></p><p class="Body-MnRsT Paragraph-cbzGIL bAebkj">- J'adore le podcast politique d'Ezra Klein, et <a href="https://itunes.apple.com/gb/podcast/the-ezra-klein-show/id1081584611?mt=2&amp;i=1000428850914#" rel="nofollow">le dernier épisode parle d'une révolution dans la manière de concevoir la misogynie</a>. (Je ne l'ai pas encore écouté.)</p><p class="Body-MnRsT Paragraph-cbzGIL bAebkj">- Si vous voulez grincer des dents, <a href="https://slate.com/human-interest/2019/01/anal-sex-dislike-advice-how-to-do-it-stoya.html" rel="nofollow">lisez cette chronique "conseil" de Slate</a>, dédiée à une femme dont le copain insiste pour la sodomiser alors qu'elle déteste ça (elle finit en larmes à tous les coups, la catastrophe totale, et pourtant il veut toujours recommencer). Il manque juuuuste une petite formulation que j'aurais personnellement soulignée en rouge fluo : "enfoiré de sociopathe manipulateur".</p><p class="Body-MnRsT Paragraph-cbzGIL bAebkj">- On a les news qu'on mérite : <a href="https://dfw.cbslocal.com/2019/01/28/rams-male-cheerleaders-super-bowl/" rel="nofollow">il y aura deux pom-pom-boys au Superbowl</a>.</p><p class="Body-MnRsT Paragraph-cbzGIL bAebkj"><strong>Recommandation culturelle :</strong></p><p class="Body-MnRsT Paragraph-cbzGIL bAebkj">J'adore <a href="https://twitter.com/Reductress" rel="nofollow">le compte Twitter Reductress</a> qui parodie les titres des magazines féminins. Mes préférés : "comment jouer de la basse suffisamment bien pour avoir le niveau d'une bassiste femme", "comment choisir le meilleur restaurant pour s'engueuler avec sa mère", "comment quitter son job parce que vos collègues n'ont pas remarqué votre nouvelle coupe de cheveux", "le saviez-vous : vos tétons font la même taille que votre anus" ou encore "désolée, les féministes : les hommes sont bien meilleurs en meurtre que nous".</p><div class="AdWrapperRow-ewuCWT knVSff"><div data-test-id="AdBannerWrapper" class="Hide-fmACZo AdBannerWrapper-fbkyjt jUzCdk"><div class="sc-bwzfXH bGgvvP"><div style="height:auto" class="sc-bdVaJa eWSIPO"><div class="ad " data-ad-unit-path="/5574/fr-gq-cp/sexe/sexactu/article/display-pos-4" aria-hidden="true" data-size-mapping="[{&quot;viewport&quot;:[0,0],&quot;slot&quot;:[[2,2],[300,50],[300,100],[300,250],[300,600],[320,50],[320,100],[320,160],[336,280]],&quot;initialHeight&quot;:600},{&quot;viewport&quot;:[900,0],&quot;slot&quot;:[],&quot;initialHeight&quot;:0}]" data-kv-pairs="[[&quot;display-pos&quot;,4]]" id="ad-25"></div></div></div></div></div></section><div class="FlexVerticalWrap-fZCTSD StickyFlexVerticalWrap-fuJJzh AdsSidebar-zrBFy dJemIW"><div class="StickyWrapper-fvWsaY StickyAdWrapper-iZThFA PIYqM"><div class="sc-bwzfXH bGgvvP"><div style="height:auto" class="sc-bdVaJa bbrWPV"><div class="ad " data-ad-unit-path="/5574/fr-gq-cp/sexe/sexactu/article/display-pos-2" aria-hidden="true" data-size-mapping="[{&quot;viewport&quot;:[0,0],&quot;slot&quot;:[],&quot;initialHeight&quot;:0},{&quot;viewport&quot;:[900,0],&quot;slot&quot;:[[120,600],[160,600],[200,600],[300,250],[300,600]],&quot;initialHeight&quot;:600}]" data-kv-pairs="[[&quot;display-pos&quot;,2]]" id="ad-26"></div></div></div></div><div class="StickyWrapper-fvWsaY StickyAdWrapper-iZThFA PIYqM"><div class="sc-bwzfXH bGgvvP"><div style="height:auto" class="sc-bdVaJa bbrWPV"><div class="ad " data-ad-unit-path="/5574/fr-gq-cp/sexe/sexactu/article/display-pos-3" aria-hidden="true" data-size-mapping="[{&quot;viewport&quot;:[0,0],&quot;slot&quot;:[],&quot;initialHeight&quot;:0},{&quot;viewport&quot;:[900,0],&quot;slot&quot;:[[120,600],[160,600],[200,600],[300,250],[300,600]],&quot;initialHeight&quot;:600}]" data-kv-pairs="[[&quot;display-pos&quot;,3]]" id="ad-27"></div></div></div></div><div class="StickyWrapper-fvWsaY StickyAdWrapper-iZThFA PIYqM"><div class="sc-bwzfXH bGgvvP"><div style="height:auto" class="sc-bdVaJa bbrWPV"><div class="ad " data-ad-unit-path="/5574/fr-gq-cp/sexe/sexactu/article/display-pos-4" aria-hidden="true" data-size-mapping="[{&quot;viewport&quot;:[0,0],&quot;slot&quot;:[],&quot;initialHeight&quot;:0},{&quot;viewport&quot;:[900,0],&quot;slot&quot;:[[120,600],[160,600],[200,600],[300,250],[300,600]],&quot;initialHeight&quot;:600}]" data-kv-pairs="[[&quot;display-pos&quot;,4]]" id="ad-28"></div></div></div></div></div></div></div>

So unfortunatly no, the CSS selector won't help.
Thank you anyway !

Reply:
We are talking about classes like 

- `ArticleWrapper-liwRXn`,
- `Body-MnRsT` and
- `Paragraph-cbzGIL`,

right?

All of those can be captured using CSS selectors. What am I missing?

Reply:
class names are different each time the GQ application is redeployed (seems like their web app framework performs some "optimizations" for "CEO" or things like that ...

Reply:
Cruelty! :cry: